lib/much-result/transaction.rb

```diff
@@ -17,7 +17,9 @@ def result
 
   def call(&block)
     begin
-      @receiver.transaction { block.call(self) }
+      @receiver.transaction do
+        catch(:halt) { block.call(self) }
```

Comment:
@kellyredding - You might use a more unique name to avoid collisions? I wanna say Sinatra or somebody uses `:halt`. Maybe `:muchresult_halt`?